### PR TITLE
PR to make QgsVectorLayerSaveAsDialog available in PyQGIS

### PR DESCRIPTION
**Pull request by jakimowb**
## Description

This PR makes the QgsVectorLayerSaveAsDialog available in the public API / PyQGIS
A new python unittest is provied in `/tests/src/python/test_qgsvectorlayersaveasdialog.py `

grafik

### CODE REVIEW

File: python/gui/auto_generated/ogr/qgsvectorlayersaveasdialog.sip.in [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+Refer to the OGR documentation for the target format for available options.
+%End
+
+    QStringList layerOptions() const;
+%Docstring
+Returns a list of additional layer options which are passed to OGR.
+Refer to the OGR documentation for the target format for available options.
+%End
+
+ long crs() const;
```
Comment (rouault) [inline python/gui/auto_generated/ogr/qgsvectorlayersaveasdialog.sip.in:87]: This method should be excluded with SIP_SKIP

File: python/gui/auto_generated/ogr/qgsvectorlayersaveasdialog.sip.in [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+      SelectedOnly,
+      GeometryType,
+      Extent,
+      Metadata,
+      AllOptions
+    };
+    typedef QFlags<QgsVectorLayerSaveAsDialog::Option> Options;
+
+
+ QgsVectorLayerSaveAsDialog( long srsid, QWidget *parent = 0, Qt::WindowFlags fl = Qt::WindowFlags() );
```
Comment (rouault) [inline python/gui/auto_generated/ogr/qgsvectorlayersaveasdialog.sip.in:40]: This method should be excluded with SIP_SKIP

**Comment by jakimowb** [comment]
Dear @rouault, as you have recommended, I skipped `long crs()` and the deprecated constructor. 
Is there anything else I can do to support a merge?

**Comment by rouault** [comment]
Looks good to me. I'd prefer if other QGIS devs could confirm it is a good idea

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**